src/js/components/FragmentList/FragmentList.jsx

```diff
@@ -72,14 +72,22 @@ const sortingOptions = {
 
 const FragmentList = ({ tabId }) => {
   const data = useSelector((state) => state.pageData[tabId].fragments);
-  const parsedData = mapDataToComponents(data, tabId);
+
+  let parsedData = mapDataToComponents(data, tabId);
   const [fragments, setFragments] = useState(parsedData);
   const [currentSorting, setCurrentSorting] = useState(null);
 
   const typeSortComparator = (a, b) => a.props.type.localeCompare(b.props.type);
   const idSortComparator = (a, b) => a.props.id.localeCompare(b.props.id);
   const timeSortComparator = (a, b) => a.props.time - b.props.time;
 
+  useEffect(() => {
+    parsedData = mapDataToComponents(data, tabId);
+    console.log(parsedData);
+    setFragments(parsedData);
```

Comment:
I think this should be seperate function that sets fragment to parsedData and is resetting sort as this is used like 5 times inside this code